opm/input/eclipse/Deck/FileDeck.cpp

```diff
@@ -139,127 +186,151 @@ FileDeck::FileDeck(const Deck& deck)
     block.erase(index);
 }
 
-void FileDeck::erase(const Index& begin, const Index& end) {
+void FileDeck::erase(const Index& begin, const Index& end)
+{
     auto current = end;
     while (current != begin) {
         --current;
         this->erase(current);
     }
 }
 
-bool FileDeck::Block::empty() const {
+bool FileDeck::Block::empty() const
+{
     return this->keywords.empty();
 }
 
-void FileDeck::Block::erase(const FileDeck::Index& index) {
-    if (index.keyword_index >= this->keywords.size())
+void FileDeck::Block::erase(const FileDeck::Index& index)
+{
+    if (index.keyword_index >= this->keywords.size()) {
         throw std::logic_error("Invalid keyword index in block");
+    }
 
     this->keywords.erase(this->keywords.begin() + index.keyword_index);
 }
 
-void FileDeck::Block::insert(std::size_t keyword_index, const DeckKeyword& keyword) {
+void FileDeck::Block::insert(const std::size_t keyword_index,
+                             const DeckKeyword& keyword)
+{
     this->keywords.insert(this->keywords.begin() + keyword_index, keyword);
 }
 
-void FileDeck::Block::dump(DeckOutput& out) const {
+void FileDeck::Block::dump(DeckOutput& out) const
+{
     for (const auto& kw : this->keywords) {
-        kw.write( out );
-        out.write_string( out.fmt.keyword_sep );
+        kw.write(out);
+        out.write_string(out.fmt.keyword_sep);
     }
 }
 
-
 FileDeck::FileDeck::Block::Block(const std::string& filename)
     : fname(fs::canonical(filename).generic_string())
 {}
 
-std::optional<FileDeck::Index> FileDeck::find(const std::string& keyword, const Index& offset) const {
+std::optional<FileDeck::Index>
+FileDeck::find(const std::string& keyword, const Index& offset) const
+{
     std::size_t file_index = offset.file_index;
     std::size_t keyword_index = offset.keyword_index;
 
     while (true) {
-        if (file_index >= this->blocks.size())
+        if (file_index >= this->blocks.size()) {
             break;
+        }
 
         const auto& file_block = this->blocks[file_index];
         const auto& block_index = file_block.find(keyword, keyword_index);
-        if (block_index.has_value())
+        if (block_index.has_value()) {
             return std::make_optional<FileDeck::Index>(file_index, block_index.value(), this);
+        }
 
-        file_index++;
+        ++file_index;
         keyword_index = 0;
     }
 
     return {};
 }
 
-std::optional<FileDeck::Index> FileDeck::find(const std::string& keyword) const {
+std::optional<FileDeck::Index>
+FileDeck::find(const std::string& keyword) const
+{
     return this->find(keyword, this->start());
 }
 
-std::size_t FileDeck::count(const std::string& keyword) const {
+std::size_t FileDeck::count(const std::string& keyword) const
+{
     std::size_t c = 0;
+
     auto index = this->start();
     while (index != this->stop()) {
-        const auto& deck_kw = this->operator[](index);
-        if (deck_kw.name() == keyword)
+        const auto& deck_kw = (*this)[index];
```

Comment:
could be std::count_if ?

Reply:
> could be `std::count_if`?

Not directly out of the box, sadly, although I agree that would be a laudable goal.  The `FileDeck::Index` type returned by `FileDeck::start()` and `FileDeck::stop()` does not meet the requirements of an iterator but I **believe** it could be made to do so with a little work.  Maybe we could do so in a follow-up PR?